packages/fxa-auth-server/lib/routes/account.js

```diff
@@ -1423,21 +1423,7 @@ module.exports = (
                   }
 
                   if (config.subscriptions && config.subscriptions.enabled) {
-                    // TODO: We should probably delete the upstream customer for
-                    // subscriptions, but no such API exists in subhub yet.
-                    // https://github.com/mozilla/subhub/issues/61
-
-                    // Cancel all subscriptions before deletion, if any exist
-                    // Subscription records will be deleted from DB as part of account
-                    // deletion, but we have to trigger cancellation in payment systems
-                    const uid = emailRecord.uid;
-                    const subscriptions = await db.fetchAccountSubscriptions(
-                      uid
-                    );
-                    for (const subscription of subscriptions) {
-                      const { subscriptionId } = subscription;
-                      await subhub.cancelSubscription(uid, subscriptionId);
-                    }
+                    await subhub.deleteCustomer(emailRecord.uid);
```

Comment:
Still getting up to speed on this repo, so forgive my naive questions--just trying to understand how this all fits together.

I'm trying to understand what happens if this line [throws](https://github.com/mozilla/fxa/pull/1671/files#diff-cc3fa280c8b38abc492139b546e1f9daR313) (say, due to a 4xx or 5xx error with subhub). I think this inner async function would return a rejected Promise, but this chain doesn't have a rejection handler, and I don't see how a rejected Promise would be handled by the next outer function, either. Is there implicit error handling somewhere? 

I noticed a sort-of related [test](https://github.com/mozilla/fxa/blob/master/packages/fxa-auth-server/test/local/routes/subscriptions.js#L507) that exercises the `subhub.cancelSubscription()` failed case, but it doesn't look like it was updated here. Would it be good to add new tests for `subhub.deleteCustomer`, including failing because of returned 4xx or 5xx errors, or are those tests somewhere else, and I haven't found them yet, or is that just not needed?

I wonder if it might be easier to reason about this handler if the whole thing were converted to await style, instead of chaining / nesting promises with awaits sprinkled in--or maybe I'll get used to it eventually :-)

Reply:
> I wonder if it might be easier to reason about this handler if the whole thing were converted to await style, instead of chaining / nesting promises with awaits sprinkled in--or maybe I'll get used to it eventually :-)

No, you're right, it is weird and I will refactor it. The mix of styles is just from where code has grown organically and collectively we've been too lazy (or rather, it's not been prioritised) to make a wholesale switch to async/await. As you well know, some of the FxA code is pretty old. 😄

> Is there implicit error handling somewhere?

The extra error handling you're looking for is in `lib/server.js` (which is probably quite a useful module to spend a bit of time getting to know):

https://github.com/mozilla/fxa/blob/95ae176c546bdc33abe18d0f97b9dba4b98d79ef/packages/fxa-auth-server/lib/server.js#L282-L288

That in turn calls `AppError.translate`, which enters the fallback condition for returning a `500` error:

https://github.com/mozilla/fxa/blob/95ae176c546bdc33abe18d0f97b9dba4b98d79ef/packages/fxa-auth-server/lib/error.js#L196-L204

> Would it be good to add new tests for `subhub.deleteCustomer`, including failing because of returned 4xx or 5xx errors, or are those tests somewhere else, and I haven't found them yet, or is that just not needed?

Good point, I'll add a test.

Reply:
ohhhh, I see. Thanks for filling in some of the details for me!